src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -308,6 +294,26 @@ void Pane::_ControlClosedHandler()
     }
 }
 
+// Method Description:
+// - Called when our terminal changes its font size or sets it for the first time
+//   (because when we just create terminal via its ctor it has invalid font size).
+//   On the latter event, we tell the root pane to resize itself so that its
+//   descendants (including ourself) can properly snap to character grids. In future,
+//   we may also want to do that on regular font changes.
+// Arguments:
+// - fontWidth - new font width in pixels
+// - fontHeight - new font height in pixels
+// - isInitialChange - whether terminal just got its proper font size.
+// Return Value:
+// - <none>
+void Pane::_FontSizeChangedHandler(const int /* fontWidth */, const int /* fontHeight */, const bool isInitialChange)

```

Comment:
It looks like this is the only place where `_rootPane` is used - is there another way we can bubble this information up somehow? I'd prefer if we avoided adding a pointer to the root if at all possible. That feels like breaking the abstraction of the tree, you know?

Reply:
Of course I could store pointer to parent instead of root and bubble the information up to the root, but that then honestly feels like extra complexity (ep. with keeping track of parent) without practical benefit. Please decide though.

Reply:
It might be terribly annoying, but would it be possible to do this as an event like `ConnectionClosed`, and have it bubble up to the Tab, which then calls `ResizeContent` on the root pane? I just really don't love the idea of having a pointer to somewhere higher on the tree.

Or __even better__, we could just have the `Tab` register a handler for each child control's `FontSizeChanged` event. Each `Split` goes through the Tab anyways, so the `Tab` has a chance to add event handlers then, and `Tab` is the layer that owns the `_root` Pane anyways.